coil-base/src/main/java/coil/size/ViewSizeResolver.kt

```diff
@@ -34,34 +34,26 @@ interface ViewSizeResolver<T : View> : SizeResolver {
     val view: T
 
     /** If true, the [view]'s padding will be subtracted from its size. */
-    val subtractPadding: Boolean
-        get() = true
+    val subtractPadding: Boolean get() = true
 
     override suspend fun size(): Size {
-        // Fast path: we don't need to wait for the view to be measured.
-        val isLayoutRequested = view.isLayoutRequested
-        val width = getWidth(isLayoutRequested)
-        val height = getHeight(isLayoutRequested)
-        if (width > 0 && height > 0) {
-            return PixelSize(width, height)
-        }
+        // Fast path: the view is already measured.
+        getSize(view.isLayoutRequested)?.let { return it }
 
-        // Wait for the view to be measured.
+        // Slow path: wait for the view to be measured.
         return suspendCancellableCoroutine { continuation ->
             val viewTreeObserver = view.viewTreeObserver
 
             val preDrawListener = object : ViewTreeObserver.OnPreDrawListener {
                 private var isResumed = false
 
                 override fun onPreDraw(): Boolean {
-                    if (!isResumed) {
-                        isResumed = true
+                    if (isResumed) return true
 
+                    val size = getSize(false)
+                    if (size != null) {
+                        isResumed = true
```

Comment:
`isResumed` becomes unnecessary because the listener is removed at the same time, meaning `onPreDraw` will never be called again.

Reply:
Good point - I remember adding it a while back for safety, but considering `androidx.core.view.OneShotPreDrawListener` doesn't guard against `onPreDraw` being called again after removing the listener it's probably safe to remove.